sam-react-app/coreconcept/src/moviesByGenre/MoviesByGenre.test.tsx

```diff
@@ -1,5 +1,40 @@
 import React from 'react';
+import {render, fireEvent, screen} from '@testing-library/react';
+import MoviesByGenre from "./MoviesByGenre";
 
 describe('SearchFormTest', () => {
+    const testGenreList = ["option a","option b","option c"];
 
+    it('renders all genres as options correctly', () => {
+        render(<MoviesByGenre
+            genreList={testGenreList}
+            selectedGenre={testGenreList[1]}
+            onSelect={()=>{}} />);
+        const selectElement = screen.getByRole('select');
+
+        expect(Array.from(selectElement.children).map(option =>option.getAttribute('value')).join(",")).toEqual('option a,option b,option c');
+    });
+
+    it('renders option b as a selected genre correctly', () => {
+        render(<MoviesByGenre
```

Comment:
You can extract component render into a function that accepts components props and renders the component itself.
```
const defaultProps: ComponentProps = {
  foo: 'bar',
  a: 1,
}
const renderComponent = (props: ComponentProps) => {
  <MoviesByGenre {...props} />
}
it('tests', () => {
  const props = {
    ...defaultProps,
    a: 2
  };
  renderComponent(props);
});
```

Reply:
done